### ISSUE TITLE
Python Packaging Setup

### ISSUE THREAD

**Issue by kb1lqc** [FaradayRF/faradayio#1]
Starting off by packaging the project from the start. Since we're starting with python 3 I'm using the following guides:

- http://www.diveintopython3.net/packaging.html
- https://packaging.python.org/tutorials/distributing-packages/
- https://docs.python.org/3/library/distribution.html

- [x] Ensure the package is built correctly (installed python modules), there is no program to run yet. However, unit testing should accomplish this.
- [x] Automate package pushing to the `pypi` servers

**Comment by kb1lqc** [comment]
Cool, we can create windows installation programs!

![image](https://user-images.githubusercontent.com/331540/33418293-0d1c6bf2-d55a-11e7-8f47-db06c307fcf3.png)

**Comment by kb1lqc** [comment]
Oh wow we can automate the python package deployment! https://docs.travis-ci.com/user/deployment/pypi/

**Comment by kb1lqc** [comment]
I can see that my locally generated source distribution can install and be imported with `from faradayio import faraday`

```
>>> from faradayio import faraday
>>> dir(faraday)
['Faraday', 'Monitor', 'TunnelServer', '__builtins__', '__cached__', '__doc__', '__file__', '__loader__', '__name__', '__package__', '__spec__', 'pytun', 'sliplib', 'threading']
```

This appears to have the proper `Faraday`, `Monitor`, and `TunnelServer` classes available as expected!

**Comment by kb1lqc** [comment]
I have [Travis-CI encryption](https://docs.travis-ci.com/user/deployment/pypi/) working on my computer after installing `ruby`. However. I'm concerned that I need to perform the encrypt command when working in a faradayrf checked out repository. This may work if I encrypt it with my personal `kb1lqc/faradayio` repository but when we want to upload it from the FaradayRF repo it may fail to decode since Travis-CI associates each repo with a different public/private key.

Is this true @reillyeon @hdkmike @el-iso @lqdev?

**Comment by kb1lqc** [comment]
We will see how this goes. I manually made travis encrypt use `faradayrf/faradayio` as the repository so hopefully it signed with that private key.

**Comment by kb1lqc** [comment]
Closing since the main intent of this ticket has been achieved and #32 covers actual deployments from Travis CI.

I have performed an automated upload attempt on [build#86](https://travis-ci.org/kb1lqc/faradayio/builds/335048012). It appears that the deployment was attempted but halted since the deployment may have been required to be from a faradayrf repository branch. The following output were the last two lines in the build:
```
Skipping a deployment with the pypi provider because this branch is not permitted
Done. Your build exited with 0.
```

This appears to have at least triggered an automated deployment of an otherwise good module.
